### DISCUSSION TITLE
persist middleware saving store as undefined in localstorage

### DISCUSSION THREAD

**Discussion by 0xdeployer** [pmndrs/zustand#2369] [Bug report]
Not exactly sure what's causing this yet, but I've named my store app-store in persist middleware and I'm seeing undefined saved to localstorage in addition to app store. For testing i am generating new user IDs if one does not already exist. I was seeing an issue with new IDs being generated when they should be persisting in local storage. 

![image](https://github.com/pmndrs/zustand/assets/87840072/b274258f-5294-4fbb-a92a-f7cd23566233)

**Comment by dbritto-dev** [comment]
@0xdeployer would you mind sharing us a mininal repro/demo on stacktblitz?
